modules/auth/persistence/cms_temporal_user_storage_sql.e

```diff
@@ -0,0 +1,388 @@
+			l_parameters: STRING_TABLE [detachable ANY]
+		do
+			error_handler.reset
+			sql_begin_transaction
+			write_information_log (generator + ".remove_activation")
+			create l_parameters.make (1)
+			l_parameters.put (a_token, "token")
+			sql_modify (sql_remove_activation, l_parameters)
+			sql_commit_transaction
+			sql_finalize
+		end
+
+	delete_user (a_user: CMS_USER)
+			-- Delete user `a_user'.
+		local
+			l_parameters: STRING_TABLE [detachable ANY]
+		do
+			error_handler.reset
+			sql_begin_transaction
+			write_information_log (generator + ".delete_user")
+			create l_parameters.make (1)
+			l_parameters.put (a_user.id, "uid")
+			sql_modify (sql_delete_temp_user, l_parameters)
+			sql_commit_transaction
+			sql_finalize
+		end
+feature  {NONE} -- Implementation
+
+	last_inserted_temp_user_id: INTEGER_64
+			-- Last insert user id.
+		do
+			error_handler.reset
+			write_information_log (generator + ".last_inserted_temp_user_id")
+			sql_query (sql_last_insert_temp_user_id, Void)
+			if not sql_after then
+				Result := sql_read_integer_64 (1)
+				sql_forth
+				check one_row: sql_after end
+			end
+			sql_finalize
+		end
+
+	last_inserted_user_id: INTEGER_64
+			-- Last insert user id.
+		do
+			error_handler.reset
+			write_information_log (generator + ".last_inserted_user_id")
+			sql_query (sql_last_insert_user_id, Void)
+			if not sql_after then
+				Result := sql_read_integer_64 (1)
+				sql_forth
+				check one_row: sql_after end
+			end
+			sql_finalize
+		end
+
+feature {NONE} -- SQL select
+
+	sql_last_insert_temp_user_id: STRING = "SELECT MAX(uid) FROM auth_temp_users;"
+
+
+	Select_user_auth_temp_by_id: STRING = "SELECT uid, name, password, salt, email, application FROM auth_temp_users as u where uid=:uid;"
+
+
+	sql_insert_temp_user: STRING = "INSERT INTO auth_temp_users (name, password, salt, email, application) VALUES (:name, :password, :salt, :email, :application);"
+			-- SQL Insert to add a new user.
+
+	Select_user_by_name: STRING = "SELECT uid, name, password, salt, email, application FROM auth_temp_users WHERE name =:name;"
+			-- Retrieve user by name if exists.
+
+	Select_user_by_email: STRING = "SELECT uid, name, password, salt, email, application FROM auth_temp_users WHERE email =:email;"
+			-- Retrieve user by email if exists.
+
+	Select_user_by_activation_token: STRING = "SELECT u.uid, u.name, u.password, u.salt, u.email, u.application FROM auth_temp_users as u JOIN users_activations as ua ON ua.uid = u.uid and ua.token = :token;"
+			-- Retrieve user by activation token if exist.
+
+	Sql_remove_activation: STRING = "DELETE FROM users_activations WHERE token = :token;"
+			-- Remove activation token.		
+
+	sql_delete_temp_user: STRING = "DELETE FROM auth_temp_users WHERE uid=:uid;"
+
+
+	Sql_last_insert_user_id: STRING = "SELECT MAX(uid) FROM users;"
+
+	sql_insert_user: STRING = "INSERT INTO users (name, password, salt, email, created, status) VALUES (:name, :password, :salt, :email, :created, :status);"
```

Comment:
I agree, but at the moment CMS_TEMP_USER is not part of the core, so we can move it to the core to reuse CMS_USER_API